### PR TITLE
test(amazonq): add tests for edge cases

### PR DESCRIPTION
**Pull request by Hweinstock**
## Problem
There are two edge cases discovered during testing that warrant some tests. 
1. the language server will respond without a range, causing the result to not be rendered. 
2. the language server COULD respond with a `StringValue` instead of string (currently doesn't, but valid in the type contract). 

## Solution
- add tests for these cases, and do minor refactoring to make this easier. 

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be squash-merged at release time.
- Your code changes must meet the guidelines in [CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).
- License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amazonq/src/app/inline/completion.ts [hunk 1]
```diff
@@ -246,8 +245,7 @@ export class AmazonQInlineCompletionItemProvider implements InlineCompletionItem
                 return []
             }
 
-            const start = document.validatePosition(editor.selection.active)
```
Comment (Hweinstock) [inline packages/amazonq/src/app/inline/completion.ts:249]: In non-testing scenarios, `editor.selection.active` appears to always be the same as the position passed in here.
